TeadsSDKDemo/app/src/main/java/tv/teads/teadssdkdemo/format/mediation/AdMobWebViewFragment.kt

```diff
@@ -0,0 +1,135 @@
+package tv.teads.teadssdkdemo.format.mediation
+
+import android.annotation.SuppressLint
+import android.content.res.Configuration
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import android.widget.Toast
+import com.google.android.gms.ads.*
+import kotlinx.android.synthetic.main.fragment_inread_webview.*
+import org.greenrobot.eventbus.Subscribe
+import tv.teads.adapter.admob.TeadsAdNetworkExtras
+import tv.teads.adapter.admob.TeadsAdapter
+import tv.teads.teadssdkdemo.R
+import tv.teads.teadssdkdemo.utils.BaseFragment
+import tv.teads.teadssdkdemo.utils.ReloadEvent
+import tv.teads.webviewhelper.SyncWebViewViewGroup
+import tv.teads.webviewhelper.baseView.ObservableWrapperView
+
+/**
+ * Display inRead as Banner within a WebView using AdMob Mediation.
+ * This is an exemple using Teads WebView Helper classes to display ads within a WebView content.
+ */
+class AdMobWebViewFragment : BaseFragment(), SyncWebViewViewGroup.Listener {
+
+    private lateinit var webviewHelperSynch: SyncWebViewViewGroup
+    private lateinit var adView: AdView
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////
+     * Ad view listener
+     *//////////////////////////////////////////////////////////////////////////////////////////////////
+
+    private val adListener = object : AdListener() {
+
+        override fun onAdLoaded() {
+            webviewHelperSynch.displayAd()
+        }
+
+        override fun onAdFailedToLoad(errorCode: Int) {
+            Toast.makeText(context, "Ad loading failed: onAdFailedToLoad($errorCode)",
+                           Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_inread_webview, container, false)
+    }
+
+    @SuppressLint("SetJavaScriptEnabled")
+    override fun onViewCreated(rootView: View, savedInstanceState: Bundle?) {
+        // 1. Init AdMob (could be done in your Application class)
+        MobileAds.initialize(context, ADMOB_TEADS_APP_ID)
+
+        // 2. Create AdMob view and add it to hierarchy
+        adView = AdView(context)
+        adView.adUnitId = ADMOB_TEADS_BANNER_ID
+        adView.adSize = AdSize.MEDIUM_RECTANGLE
+
+        /*
+        For a webview integration, we provide a example of tool to synchronise the ad view with the webview.
+        You can find it in the webviewhelper module. {@see SyncWebViewTeadsAdView}
+         */
+        val observableWrapperView = ObservableWrapperView(context!!, adView)
```

Comment:
We should manage a null context in ObservableWrapperView instead of trow an exception